services/dedup/badger/badger.go

```diff
@@ -40,45 +40,42 @@ func DefaultPath() string {
 	return fmt.Sprintf(`%v%v`, tmpDirPath, badgerPathName)
 }
 
-func NewBadgerDB(path string) *BadgerDB {
-	dedupWindow := config.GetReloadableDurationVar(3600, time.Second, "Dedup.dedupWindow", "Dedup.dedupWindowInS")
-
-	log := logger.NewLogger().Child("dedup")
+func NewBadgerDB(conf *config.Config, stats stats.Stats, path string) *Dedup {
+	dedupWindow := conf.GetReloadableDurationVar(3600, time.Second, "Dedup.dedupWindow", "Dedup.dedupWindowInS")
+	log := logger.NewLogger().Child("Dedup")
 	badgerOpts := badger.
 		DefaultOptions(path).
 		WithCompression(options.None).
```

Comment:
[question] Out of curiosity... did you test this with compression at all? If yes, what did you observe?

Reply:
This was an already existing code and I didn't make any changes in the existing functionality